### PR TITLE
PYTHON-4271 Skip failing serverless proxy tests

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: test/load_balancer/sdam-error-handling.json [hunk 1]
```diff
@@ -261,6 +261,7 @@
     },
     {
       "description": "errors during the initial connection hello are ignored",
+      "skipReason": "waiting on CLOUDP-202309",
```
Comment (ShaneHarvey) [inline test/load_balancer/sdam-error-handling.json:264]: Can we skip these tests only on the Serverless Proxy? They are still useful to run on all other variants.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/load_balancer/sdam-error-handling.json (hunk 1 above)
Comment (NoahStapp) [reply test/load_balancer/sdam-error-handling.json:264]: Is it possible to identify when a client context is for a serverless proxy and not just a serverless instance?

Comment (blink1073) [reply test/load_balancer/sdam-error-handling.json:264]: We'd have to feed it in as an environment variable.   Here's where we feed it through: https://github.com/mongodb/mongo-python-driver/blob/a46331d478d68fca68bb9f85302fe411373f75b6/.evergreen/config.yml#L905

Comment (NoahStapp) [reply test/load_balancer/sdam-error-handling.json:264]: Got it, there's no way to distinguish at the client level between proxy and non-proxy.

Comment (ShaneHarvey) [reply test/load_balancer/sdam-error-handling.json:264]: In that case it would still be good to keep running these tests on all non-serverless clusters.

Comment (blink1073) [reply test/load_balancer/sdam-error-handling.json:264]: Can't we skip in in the unified test runner, where we can inspect an environment variable?

Comment (NoahStapp) [reply test/load_balancer/sdam-error-handling.json:264]: Sorry let me clarify--yes we can skip these using an environment variable, I was commenting that we don't have a cleaner way of doing so.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**